app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { email.downcase! }
+  validates :name, presence: true,length: { maximum:50 }
+  VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
```suggestion
  VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
  
  validates :name, presence: true,length: { maximum:50 }
```
and move before_save under has_secure_password